### PR TITLE
Add etcd addon

### PR DESCRIPTION
**Pull request by nehayward**
# etcd
Adds an etcd to addon folder, installed via helm.

### CODE REVIEW

File: addons/etcd/Makefile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+install:
+	helm install stable/etcd-operator --name etcd-operator
+
+uninstall:
+	helm delete --purge etcd-operator	
+
+info:
+	@echo https://github.com/helm/charts/tree/master/stable/etcd-operator
```
Comment (nehayward) [inline addons/etcd/Makefile]: Thinking of adding this to helm charts without source.

Comment (diemonster) [reply addons/etcd/Makefile]: sorry, not following you there?

**Comment by nehayward** [comment]
@diemonster so we could remove the info and add a short readme. I was trying to suggest for the package that don't build from source, (prometheus, etcd) we should link to the helm chart on GitHub so they could do configuration. 

As far as configuration the cleanest way would probably add an optional param that accepts a yaml file.

`helm install --name my-release --values values.yaml stable/etcd-operator`

**Comment by diemonster** [comment]
@nehayward I like the addition of passing optional args via the YAML file. I think we do need to provide some sensible defaults though --- for instance, are these RBAC roles created as part of the helm install? https://github.com/coreos/etcd-operator/blob/master/doc/user/rbac.md

There's some other examples here that we could crib from: https://github.com/coreos/etcd-operator/blob/master/doc/user/spec_examples.md

File: addons/etcd-operator/Makefile [hunk 2]
```diff
@@ -0,0 +1,11 @@
+install:
+ifeq ($(ETCD_YAML),)
```
Comment (zpatrick) [inline addons/etcd-operator/Makefile]: Given the purpose of this repo is to be opinionated and declarative, I think we should take this part out. 
It's also inconsistent with the rest of the addons.

Comment (nehayward) [reply addons/etcd-operator/Makefile]: That's a good point, we should probably align on our opinionated etcd. I'm not sure what that is at the moment.

File: addons/etcd-operator/README.md [hunk 3]
```diff
@@ -7,3 +7,28 @@ Installs [etcd-operator](https://github.com/helm/charts/tree/master/stable/etcd-
...
+
+- Once your etcd-operator is up and running you have two options to access it. You can expose the service or access it via another pod.
+
+#### Internal Access
+```bash
+kubectl run --rm -i --tty fun --image quay.io/coreos/etcd --restart=Never -- /bin/sh
+ETCDCTL_API=3 etcdctl --endpoints http://etcd-cluster-client:2379 put foo bar
+```
+
+#### External
```
Comment (diemonster) [inline addons/etcd-operator/README.md]: [nit] add 'Access'

File: addons/etcd-operator/README.md [hunk 4]
```diff
@@ -7,3 +7,28 @@ Installs [etcd-operator](https://github.com/helm/charts/tree/master/stable/etcd-
 - The **ETCD_YAML** env is optional, but accepts a path to a **value.yaml** file for further [configuration](https://github.com/helm/charts/tree/master/stable/etcd-operator#configuration) on etcd-operator.
 
 `make install` or `make install ETCD_YAML=""`
+
+
+### Optionals
+
+- Once your etcd-operator is up and running you have two options to access it. You can expose the service or access it via another pod.
```
Comment (diemonster) [inline addons/etcd-operator/README.md]: You might want to add a line about why you would pick external versus internal, i.e. internal service discovery versus an external k/v store.

Out of curiosity, what would a good use case for *not* using a load balancer?

Comment (nehayward) [reply addons/etcd-operator/README.md]: My initial thoughts are for security reasons. If all the services using it are in the cluster there isn't a direct need to expose it. Also when you expose with AWS it's public available, so you'd have to make it private as well.

**Pull request merged**